src/scripts/main.js

```diff
@@ -1,3 +1,29 @@
 'use strict';
 
-// write your code here
+// Select the logo element
+const logo = document.querySelector('.logo');
+
+const clickPromise = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    reject('Promise was rejected!');
+  }, 3000);
+
+  logo.addEventListener('click', () => {
+    resolve('Promise was resolved!');
+    clearTimeout(timeoutId);
+  });
+});
```

Comment:
The 'click' event listener is added inside the executor function of the promise. This means that every time you create a new promise with this code, a new event listener will be added to the logo. If you only need to handle the click event once, it might be better to add the event listener outside of the promise.